### ISSUE TITLE
Cannot load HLS live stream due to error

### ISSUE THREAD

**Issue by kevinscroggins-youi** [google/shaka-player#1784]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes.

**What version of Shaka Player are you using?**
2.4.6

**Can you reproduce the issue with our latest release version?**
Yes.

**Can you reproduce the issue with the latest code from `master`?**
Yes.

**Are you using the demo app or your own custom app?**
Custom app.

**If custom app, can you reproduce the issue using our demo app?**
Yes.

**What browser and OS are you using?**
Google Chrome 71.0.3578.98 on Mac OSX 10.13.6

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
Samsung Tizen 2017 TV (UN43MU6300 Model) (3.0 Platform) w/ retail T-KTMAKUC-1250.5 Firmware

**What are the manifest and license server URIs?**

HLS Manifest: https://rcavlive.akamaized.net/hls/live/664044/cancbft/master.m3u8?hdnts=st=1548776386~exp=1548776506~acl=/hls/live/664044/cancbft*~hmac=decbed6e6be38b1f0a96d2d15c36592864c58c308285c8044e90cc654f42fdb3

**What did you do?**

Tried to load an unencrypted HLS live stream manifest.

Fiddle: https://jsfiddle.net/t4s1ko9u

**What did you expect to happen?**
I expected that the HLS live stream would load and begin playing.

I have also tested this same live stream manifest on both Tizen's built in video player on both 2017 and 2018 TVs and the HLS reference player, and it plays without error.

**What actually happened?**

The HLS live stream failed to load with `MANIFEST.HLS_COULD_NOT_PARSE_SEGMENT_START_TIME` / `4030` error on Shaka 2.4.6.

On Shaka 2.5.0-beta2 the error message appears to be different: `Shaka Error MANIFEST.HLS_REQUIRED_ATTRIBUTE_MISSING (URI)` / `4023`

I also saw additional output in the console when using Shaka 2.4.6:
```
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_dv/00008/master_dv_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_dv/00008/master_dv_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_dv/00008/master_dv_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_128/00008/master_128_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_128/00008/master_128_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_128/00008/master_128_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_128/00008/master_128_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_128/00008/master_128_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_128/00008/master_128_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_dv/00008/master_dv_01043.aac
Unable to fetch a partial HLS segment! Falling back to a full segment request, which is expensive!  Your server should support Range requests and CORS preflights. https://rcavlive.akamaized.net/hls/live/664044/cancbft/20190128T093659/master_dv/00008/master_dv_01043.aac
```

It seems to me like this is partially a manifest issue, but also something that is possibly not being handled correctly. Any help on this would be much appreciated. Thank you!

EDIT: Formatting.

**Comment by vaage** [comment]
@kevinscroggins-youi Thank you for providing sample code with [JSFiddle](https://jsfiddle.net/). When people are as considerate as you and do that, it makes it much easier for us to reproduce.

Unfortunately when I tried to run it, the player gets a different error `Failed to load resource: the server responded with a status of 403 (Forbidden)`. By chance, is the asset still available?

**Comment by kevinscroggins-youi** [comment]
@vaage No problem! That is very strange, I don't get that at all. Perhaps it's geo blocked or something. I'll attach a copy of the HLS manifest data, if that helps? If you need it hosted, let me know and I can see about sorting that out. Thanks!

[master.txt](https://github.com/google/shaka-player/files/2829601/master.txt)

**Comment by kevinscroggins-youi** [comment]
Is there any possibility that you can try to load the stream through a VPN? If you set your VPN location to anywhere in Canada (ideally Quebec or Ontario), it should work. If not, I will see about getting a separate asset to test with. Thanks!

**Comment by chrisfillmore** [comment]
@kevinscroggins-youi I am in Ontario so I took a look at your content and reproduced your issue using Shaka v2.4.6.

`HlsParser#getStartTime_` throws HLS_COULD_NOT_PARSE_SEGMENT_START_TIME because `mimeType` is `audio/aac`, which is apparently not supported (see issue #1083). @vaage does this make sense? If you have any follow-up questions let me know and I'm happy to test again.

See the state of HlsParser at the point of the error below:

![screen shot 2019-02-08 at 3 52 05 pm](https://user-images.githubusercontent.com/7190265/52505601-06d45180-2bba-11e9-9cfd-e02d4dd22484.png)

**Comment by vaage** [comment]
@chrisfillmore thank-you for taking a look at that for us.

What you found makes sense to me, and would explain the issue. Currently release 2.7 is being scoped to focus on HLS work, @joeyparrish does Raw AAC + HLS sound like a likely candidate to be included in 2.7?
